cvapipe_analysis/tests/test_shape_space.py

```diff
@@ -0,0 +1,166 @@
+import filecmp
+from pathlib import Path
+import shutil
+import numpy as np
+import pandas as pd
+from cvapipe_analysis.tools import controller, shapespace
+from cvapipe_analysis.steps.shapemode.shapemode_tools import ShapeModeCalculator
+
+config = {
+    "aggregation": {"type": ["avg"]},
+    #"appName": "cvapipe_analysis",
+    "data": {
+        "nucleus": {"alias": "NUC", "channel": "dna_segmentation", "color": "#3AADA7"}
+    },
+    "distribute": {},
+    "features": {
+        "SHE": {
+            "aliases": ["NUC"],
+            "alignment": {"align": True, "reference": "nucleus", "unique": False},
+            "lmax": 16,
+            "sigma": 2,
+        },
+        "aliases": ["NUC"],
+    },
+    "parameterization": {
+        "inner": "NUC",
+        "number_of_interpolating_points": 32,
+        "outer": "MEM",
+        "parameterize": ["RAWSTR", "STR"],
+    },
+    "preprocessing": {
+        "filtering": {"csv": "", "filter": False, "specs": {}},
+        "remove_mitotics": True,
+        "remove_outliers": True,
+    },
+    "project": {
+        "local_staging": Path(__file__).parent / "all/shape_analysis/shape_space",
+        "overwrite": True,
+    },
+    "shapespace": {
+        "aliases": ["NUC"],
+        "map_points": [-2.0, -1.5, -1.0, -0.5, 0.0, 0.5, 1.0, 1.5, 2.0],
+        "number_of_shape_modes": 8,
+        "plot": {
+            "frame": False,
+            "limits": [-150, 150, -80, 80],
+            "swapxy_on_zproj": False,
+        },
+        "removal_pct": 0.25,
+        "sorter": "NUC",
+    },
+    "structures": {
+        "lamin": [
+            "nuclear envelope",
+            "#084AE7",
+            "{'raw': (475,1700), 'seg': (0,30), 'avgseg': (0,60)}",
+        ]
+    },
+}
+
+
+def random_shcoeffs_dataframe(nrows=100):
```

Comment:
In the future would be good to extend this test to cover other aspects of cvapipe. For example, we can start by creating binary images of spheres with random radii as a proxy for cells. I think this is good for now.